backend/lcfs/web/api/document/views.py

```diff
@@ -40,14 +43,27 @@ async def get_all_documents(
     response_model=FileResponseSchema,
     status_code=status.HTTP_201_CREATED,
 )
-@view_handler([RoleEnum.SUPPLIER])
+@view_handler([RoleEnum.SUPPLIER, RoleEnum.ANALYST])
 async def upload_file(
     request: Request,
     parent_id: int,
     parent_type: str,
     file: UploadFile = File(...),
     document_service: DocumentService = Depends(),
+    compliance_report_service: ComplianceReportServices = Depends(),
 ) -> FileResponseSchema:
+    # Fetch the compliance report
+    compliance_report = await compliance_report_service.get_compliance_report_by_id(parent_id)
+    if not compliance_report:
+        raise HTTPException(status_code=404, detail="Compliance report not found")
+
+    # Check if the compliance report is submitted
+    if compliance_report.current_status.status == ComplianceReportStatusEnum.Submitted:
```

Comment:
- Can you please move this logic to service.py or validation.py?
- Restriction to upload file when ComplianceReport is not in Draft status applies only to Supplier users. Analyst should be able to upload when it's in Submitted status as per LCFS #1426 , "Enable the IDIR analyst to upload file attachments to compliance reports or supplemental reports that are in either Submitted or Assessed status. "